packages/fe-mockserver-core/src/mockdata/fileBasedMockData.ts

```diff
@@ -59,6 +59,13 @@ export class FileBasedMockData {
                                 );
                             });
                         }
+                        if (!prop.nullable && !mockLine.hasOwnProperty(prop.name)) {
```

Comment:
Definitely makes sense. I wonder about the handling of ComplexTypes above, though. There can be something like

```
<Property Name="blabla" Type="BlaBlaType" />

<ComplexType Name="BlaBlaType">
   <Property Name="primitiveComponent" Type="Edm.Byte" Nullable="false" />
   …
</ComplexType>

```

And in that case, `blabla` may be null, but if it is not, then `blabla.primitiveComponent` must not be null. But that is maybe something to check independently, or leave it as it is if it works ok in practice.